### PR TITLE
Add tests for activate_account view

### LINKED ISSUE #5333: Add tests for User Auth & Password Reset module
### Description

Part of #4480 

I would like to contribute to improving test coverage for the User Auth & Password Reset module.

Plan:
- Start with 1 file and expand gradually
- Cover authentication flows and edge cases

Please let me know if this works 

### Steps to Reproduce

_No response_

### Expected Behavior

_No response_

### Actual Behavior

_No response_

### Screenshots

_No response_

### Operating System

_No response_

### Browser

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by celesmedina**
## Description
  The following cases are covered:                                                                                                                                                                                                         
  - Valid key activates the user account                                                                                                                                                                                                   
  - Valid key strips the activation key suffix from the stored password                                                                                                                                                                    
  - Valid key redirects to the profile page after activation                                                                                                                                                                               
  - Missing username in GET params returns a 500 error response                                                                                                                                                                            
  - Missing key in GET params returns a 500 error response                                                                                                                                                                                 
  - Nonexistent username returns a 500 error response                                                                                                                                                                                      
  - Already active account returns a 500 error response                                                                                                                                                                                    
  - Wrong activation key returns a 500 error response                                                                                                                                                                                      
                                                                                                                                                                                                                                           
  ## Related Issue                                                                                                                                                                                                                       
                                                                                                                                                                                                                                           
  Closes #5333                                                                                                                                                                                                                           

  ## Type of Change

  - [ ] Bug fix
  - [ ] New feature
  - [ ] Breaking change
  - [ ] Documentation update
  - [x] Refactor / cleanup
  - [ ] CI/CD or build change                                                                                                                                                                                                              
  
  ## Testing                                                                                                                                                                                                                               
                                                                                                                                                                                                                                         
  - [x] Existing tests pass
  - [x] New tests added (if applicable)
  - [x] Manually verified                                                                                                                                                                                                                  
                                                                                                                                                                                                                                           
  ## AI Disclosure                                                                                                                                                                                                                         
                                                                                                                                                                                                                                           
  Used AI assistance (Claude) to help structure and draft the tests. All tests were reviewed and verified by me.                                                                                                                           
  
  ## Checklist                                                                                                                                                                                                                             
                                                                                                                                                                                                                                         
  - [x] Self-reviewed the code                                                                                                                                                                                                             
  - [ ] Updated documentation (if needed)
  - [x] No new warnings or errors introduced

### CODE REVIEW

File: esp/esp/users/tests.py [hunk 1]
```diff
@@ -1099,3 +1100,71 @@ def testEmptyGuardianEmail(self):
         email_errors = [e for e in form.non_field_errors()
                         if 'email' in e.lower()]
         self.assertEqual(email_errors, [])
+
+
+class ActivateAccountTest(TestCase):
+    """Tests for the activate_account view (part of #4480)."""
```
Comment (Oval17) [inline esp/esp/users/tests.py]: Docstring says "part of #4480" but this PR closes #5333. Either reference #5333 or both,
if #4480 is the parent tracking issue.

**Review by Oval17 — APPROVED** [review]
Tests are correct and well-structured. The setUp accurately mirrors the view's password storage format (password + '_' + key) from what I can see. 
Good to merge.

File: esp/esp/users/tests.py (hunk 1 above)
Comment (Oval17) [reply esp/esp/users/tests.py]: Ignore this if Parent ticket is part of GSoC Tag.

Comment (willgearty) [reply esp/esp/users/tests.py]: We don't need to reference issues or PRs in comments in the codebase.

Comment (celesmedina) [reply esp/esp/users/tests.py]: Removed

**Comment by aryansondharva** [comment]
Hi @willgearty,

I'm Aryan (first-year student). I want to make my first contribution by adding unit tests for one small file.
I would like to work on esp/esp/users/tests.py from the "Other Modules" / "Medium" category.
I will create a new sub-issue for it and start with basic tests using ProgramFrameworkTest.

Is this okay? Thank you!

**Comment by aryansondharva** [comment]
Nice test coverage. One question: would it also be useful to add a test verifying that an activation key cannot be reused after a successful activation?

**Comment by kkbrum** [comment]
@Oval17 could you fix the merge conflict and then I'll get this merged in?

**Review by Oval17 — APPROVED** [review]
LGTM
